### PR TITLE
AuditEvent Resource 

### LINKED ISSUE #51: [Quest/Malawi Core] Audit-event Resource
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by evance-mose**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #[issue number] 

**Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/engine/src/main/java/org/smartregister/fhircore/engine/auditEvent/AuditEventWorker.kt [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    private suspend fun createAuditEvent(){
+        // Get Practitioner Resource
+        val practitionerID =
+            sharedPreferences.read(key = SharedPreferenceKey.PRACTITIONER_ID.name, defaultValue = null)
+        val practitioner = defaultRepository.loadResource<Practitioner>(practitionerID!!)
+
+        // Create AuditEvent Resource
+        val auditEvent = AuditEvent().apply {
```
Comment (sevenreup) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/auditEvent/AuditEventWorker.kt]: Let's have an AuditRepository that has the code for creating the audit events. You can inject that and use it

**Review by sevenreup — CHANGES_REQUESTED** [review]

**Review by sevenreup — APPROVED** [review]
